src/key_value_store.js

```diff
@@ -320,8 +321,8 @@ export class KeyValueStore {
 
         const response = await keyValueStores.listKeys({ storeId: this.storeId, exclusiveStartKey });
         const { nextExclusiveStartKey, isTruncated, items } = response;
-        for (const key of items) {
-            await iteratee(key, index++);
+        for (const item of items) {
+            await iteratee(item.key, item.size, index++);
```

Comment:
I am thinking about parameter order .. maybe the index will be used more often than size.  And one more thing - looking into AWS S3 docs, we will be able to also later add modification date. So maybe the best would be to have function signature:

```javascript
const iteratee = function (key, index, info = { size, modifiedAt }) {
   ...
}
```

what do you think?

Reply:
Actually I was thinking the same, to swap index and size.

Reply:
... and wrap size in info object to make it future proof :)